### PR TITLE
Add point in time api

### LINKED ISSUE #251: [FEATURE] Add point_in_time API
### Is your feature request related to a problem?
Currently the lib does not support the `point_in_time api` used for pagination searching.
https://opensearch.org/docs/latest/search-plugins/point-in-time-api/

### What solution would you like?
Implement the new API Endpoints described in the docs.

### PR DESCRIPTION
**Pull request by Jakob3xD**
### Description
Add  the Point in Time API.
https://opensearch.org/docs/latest/search-plugins/point-in-time-api/

### Issues Resolved
Closes #251 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by Jakob3xD** [comment]
@dblock does this lib aim to be a low level lib or should it be more user friendly?
I am asking because I added some custom Response types for Point In Time requests and also avoid the user to build the delete body.

**Comment by dblock** [comment]
> @dblock does this lib aim to be a low level lib or should it be more user friendly? I am asking because I added some custom Response types for Point In Time requests and also avoid the user to build the delete body.

I definitely think we want to minimize the amount of work a user has to do. So as much user friendly as possible, but also letting users go deep into low level constructs.

File: opensearchapi/api.pointintime.create.go [hunk 1]
```diff
@@ -0,0 +1,260 @@
+// SPDX-License-Identifier: Apache-2.0
+//
+// The OpenSearch Contributors require contributions made to
+// this file be licensed under the Apache-2.0 license or a
+// compatible open source license.
+//
+// Modifications Copyright OpenSearch Contributors. See
+// GitHub history for details.
+
+// Licensed to Elasticsearch B.V. under one or more contributor
```
Comment (dblock) [inline opensearchapi/api.pointintime.create.go:10]: This is new code? We don't need a ES grant, remove this license part below.

**Review by dblock — APPROVED** [review]
Small nit.

File: opensearchapi/api.pointintime.create.go (hunk 1 above)
Comment (Jakob3xD) [reply opensearchapi/api.pointintime.create.go:10]: Question for me as I am not into such license things.
Can I remove the ES grant if it is new code but follows the same scheme and has similarity to code that has ES grant? As the `Withxyz()` functions are all the same.

Comment (dblock) [reply opensearchapi/api.pointintime.create.go:10]: IANAL, but if you _copied_ any code with such license then we should probably keep it. But if it's _new code_ then we don't need it. Check out https://github.com/opensearch-project/.github/blob/main/CONTRIBUTING.md#license-headers for more info.

**Pull request merged**